### PR TITLE
Add update checker

### LINKED ISSUE #2333: Add alteryx_open_src_update_checker as an extras_require package
- The alteryx_open_src_update_checker supports EvalML and can be added to setup.py
- https://github.com/FeatureLabs/alteryx-open-src-update-checker/
- Example PR: https://github.com/alteryx/woodwork/pull/959

### PR DESCRIPTION
**Pull request by jeff-hernandez**
Closes #2333 by adding the Alteryx open-source update checker. The update checker will automatically check if there's a newer release of EvalML available.

### CODE REVIEW

File: setup.py
Comment (gsheni) [inline setup.py:6]: @jeff-hernandez lets add an install workflow and test that the update checker installs properly.
https://github.com/alteryx/featuretools/blob/main/.github/workflows/install_test.yml#L34

Comment (jeff-hernandez) [reply setup.py:6]: Added workflow

**Review by angela97lin — COMMENTED** [review]
Just to make sure I understand the purpose of this: this is supposed to check that the packages evalml depends on are up to date? From the README, it sounded like it would automatically download the latest version of evalml but I tried to purposely downgrade evalml (to 0.24.0) an run `python -m pip install evalml[update_checker]` and nothing happened?

**Comment by gsheni** [comment]
@angela97 this will only work once it is merged into main and released. In addition, it will only work with versions of EvalML going forward. The way this would work is the following.
1. Have EvalML + updated checker installed
2. New version of EvalML comes out
3. The open-src-update-checker puts a warning in the console/output when you **import evalml**

**Review by angela97lin — APPROVED** [review]
@gsheni Got it, thank you for explaining! I wonder if there's a clearer way to express this in the README, but otherwise this looks good! Thanks @jeff-hernandez for doing this 😁

**Comment by gsheni** [comment]
@jeff-hernandez One thing I just realized, we should add this to the Install page
- https://evalml.alteryx.com/en/latest/install.html

**Review by chukarsten — APPROVED** [review]

File: evalml/update_checker.py
Comment (dsherry) [inline evalml/update_checker.py:1]: @jeff-hernandez to date we've reserved the top directory level of the `evalml` folder for modules only. Could you please move this file to `evalml/utils/`?

File: README.md
Comment (dsherry) [inline README.md:26]: Docs style nit-pick: line break instead of dash
> **Update checker**
> Receive automatic notifications of new EvalML releases

File: .github/workflows/install_test.yml
Comment (dsherry) [inline .github/workflows/install_test.yml:3]: @jeff-hernandez I'm tempted to ask you to delete this, so that these tests only run on merges to `main`. But I guess these jobs will complete rather quickly, so it may be fine to just let them run.

@chukarsten @freddyaboulton tagging you RE recent discussions about our test suite and how many jobs we run on each PR.

I wish there were a way to say "run these tests on merges to `main`, and also if the update_checker.py code has been modified", but that's probably not worth the trouble.

A nice compromise could be to run one python version on PRs and the other two on merge to `main`.

File: .github/workflows/install_test.yml
Comment (dsherry) [inline .github/workflows/install_test.yml:33]: @jeff-hernandez what does this env var do and why is it necessary here?

File: .github/workflows/install_test.yml
Comment (dsherry) [inline .github/workflows/install_test.yml:26]: @jeff-hernandez why is this necessary?

File: README.md
Comment (dsherry) [inline README.md:25]: Ultra nit-pick: use #### instead of ### to be consistent with the rest of the readme

File: .github/codecov.yml
Comment (dsherry) [inline .github/codecov.yml:14]: Seems fine to me

**Review by dsherry — CHANGES_REQUESTED** [review]
@jeff-hernandez  looks great! One blocking change: please move the python code to `evalml/utils/`

I also left a comment for @freddyaboulton and @chukarsten to dig into--they've been working on our CI recently. But that's not blocking merge

File: .github/workflows/install_test.yml
Comment (gsheni) [reply .github/workflows/install_test.yml:3]: @dsherry there is a way to say "run these tests on merges to main, and also if the update_checker.py code has been modified"

```yaml
on:
  push:
    branches:
      - main
  pull_request:
    types: [opened, synchronize]
    paths:
      - 'evalml/update_checker.py'
```

File: .github/workflows/install_test.yml
Comment (gsheni) [reply .github/workflows/install_test.yml:33]: It disables the update checker API requests, since they are not needed for testing.
- https://github.com/FeatureLabs/alteryx-open-src-update-checker#disable-checker

File: .github/workflows/install_test.yml
Comment (jeff-hernandez) [reply .github/workflows/install_test.yml:26]: It helps to make the installation process more readable

File: .github/workflows/install_test.yml
Comment (dsherry) [reply .github/workflows/install_test.yml:3]: Woah. I would be a fan of us using that here! That way, if we modify this code in the future, the tests will still run in the PR :)

File: .github/workflows/install_test.yml
Comment (dsherry) [reply .github/workflows/install_test.yml:26]: That's neat. Do you think we should do this in all our CI? Is that what featuretools, woodwork and compose do?

File: .github/workflows/install_test.yml
Comment (dsherry) [reply .github/workflows/install_test.yml:33]: Got it, thanks
Perhaps worth a code comment? I'm not sure how hard it would be for a new dev to figure that out -- they'd have to know to go look at the update checker code I guess

File: .github/workflows/install_test.yml
Comment (jeff-hernandez) [reply .github/workflows/install_test.yml:26]: Yeah, it makes the package installation easier to follow. We also do this in featuretools, woodwork, and compose.

File: .github/workflows/install_test.yml
Comment (jeff-hernandez) [reply .github/workflows/install_test.yml:3]: I'll add that to the workflow

File: .github/workflows/install_test.yml
Comment (jeff-hernandez) [reply .github/workflows/install_test.yml:33]: I'll add a comment

File: evalml/update_checker.py
Comment (jeff-hernandez) [reply evalml/update_checker.py]: Yep, moved

File: README.md
Comment (dsherry) [inline README.md:29]: Mega nit-pick but elsewhere in our docs we say `pip install ...` instead of `python -m pip install ...`

File: docs/source/install.ipynb
Comment (dsherry) [inline docs/source/install.ipynb:36]: Same nit-pick as above

**Review by dsherry — APPROVED** [review]
Thanks @jeff-hernandez looks great! Left two docs nit-picks but not blocking merge

**Pull request merged**
